core/cmd/hoverfly/main.go

```diff
@@ -283,6 +282,8 @@ func main() {
 		requestCache = nil
 	}
 
+	cfg.DisableProxyAuthoriation = *disableProxyAuth
```

Comment:
It is binary decision. You are either using standard proxy authentication, which is http and https, or you are using the custom header authentication, which is https only.